libs/features/mas/src/hydrate.js

```diff
@@ -175,16 +285,40 @@ function createSpectrumSwcButton(cta, aemFragmentMapping, isOutline, variant) {
             tabIndex: 0,
             size: aemFragmentMapping.ctas.size ?? 'm',
         },
-        cta,
+        cta.innerHTML,
     );
 
-    spectrumCta.addEventListener('click', (e) => {
-        if (e.target !== cta) {
-            /* c8 ignore next 3 */
+    (async () => {
+        try {
+          await customElements.whenDefined('checkout-button');
```

Comment:
I think it is safe to assume that `checkout-button` is already defined before we hit here, if not wee need to check how mas.js is bundled.
I would avoid adding checks for web components that we know are in the bundle.